docs/framework/wcf/samples/custom-lifetime.md

```diff
@@ -1,219 +1,227 @@
-  
-```  
-public void ApplyBehavior(ServiceDescription description,   
-           ServiceHostBase serviceHostBase,   
-           Collection<DispatchBehavior> behaviors,  
-           Collection<BindingParameterCollection> parameters)  
-{  
-    CustomLifetimeLease customLease = new CustomLifetimeLease();  
-    CustomLeaseInitializer initializer =   
-                new CustomLeaseInitializer(timeout);  
-  
-    foreach (DispatchBehavior dispatchBehavior in behaviors)  
-    {  
-        dispatchBehavior.InstanceContextLifetimes.Add(customLease);  
-        dispatchBehavior.InstanceContextInitializers.Add(initializer);  
-    }  
-}  
-```  
-  
- This behavior can be added to a sample service class by annotating it with the `CustomLeaseTime` attribute.  
-  
-```  
-[ServiceBehavior(InstanceContextMode=InstanceContextMode.Shareable)]  
-[CustomLeaseTime(Timeout = 20000)]  
-public class EchoService : IEchoService  
-{  
-  //…  
-}  
-```  
-  
- When you run the sample, the operation requests and responses are displayed in both the service and client console windows. Press ENTER in each console window to shut down the service and client.  
-  
-#### To set up, build, and run the sample  
-  
-1.  Ensure that you have performed the [One-Time Setup Procedure for the Windows Communication Foundation Samples](../../../../docs/framework/wcf/samples/one-time-setup-procedure-for-the-wcf-samples.md).  
-  
-2.  To build the C# or Visual Basic .NET edition of the solution, follow the instructions in [Building the Windows Communication Foundation Samples](../../../../docs/framework/wcf/samples/building-the-samples.md).  
-  
-3.  To run the sample in a single- or cross-machine configuration, follow the instructions in [Running the Windows Communication Foundation Samples](../../../../docs/framework/wcf/samples/running-the-samples.md).  
-  
+> Using thread locking is not a recommended approach because serialization can severely affect the performance of your application.
+
+A private member variable is used in the `CustomLifetimeLease` class to track the `IsIdle` value. Each time the value of <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> is retrieved, the `IsIdle` private member is returned and reset to `false`. It is essential to set this value to `false` in order to make sure the Dispatcher calls the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.NotifyIdle%2A> method.
```

Comment:
A private member field is used in the `CustomLifetimeLease` class to track the idle state and is returned by the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle> method. Each time the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle> method is called, the `isIdle` field is returned and reset to `false`.  It is essential to set this value to `false` in order to make sure the Dispatcher calls the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.NotifyIdle> method.